lua/autorun/server/boom_headshot.lua

```diff
@@ -1,85 +1,68 @@
 if SERVER then
-	AddCSLuaFile( 'effects/headshot.lua' )
-	AddCSLuaFile( 'effects/bloodstream.lua' )
-	AddCSLuaFile( 'autorun/client/boom_headshot.lua' )
-	resource.AddFile( 'materials/fwkzt/sprite_bloodspray1.vmt' )
-	resource.AddFile( 'materials/fwkzt/sprite_bloodspray2.vmt' )
-	resource.AddFile( 'materials/fwkzt/sprite_bloodspray3.vmt' )
-	resource.AddFile( 'materials/fwkzt/sprite_bloodspray4.vmt' )
-	resource.AddFile( 'materials/fwkzt/sprite_bloodspray5.vmt' )
-	resource.AddFile( 'materials/fwkzt/sprite_bloodspray6.vmt' )
-	resource.AddFile( 'materials/fwkzt/sprite_bloodspray7.vmt' )
-	resource.AddFile( 'materials/fwkzt/sprite_bloodspray8.vmt' )
+    AddCSLuaFile( "effects/headshot.lua" )
+    AddCSLuaFile( "effects/bloodstream.lua" )
+    AddCSLuaFile( "autorun/client/boom_headshot.lua" )
+    resource.AddFile( "materials/fwkzt/sprite_bloodspray1.vmt" )
+    resource.AddFile( "materials/fwkzt/sprite_bloodspray2.vmt" )
+    resource.AddFile( "materials/fwkzt/sprite_bloodspray3.vmt" )
+    resource.AddFile( "materials/fwkzt/sprite_bloodspray4.vmt" )
+    resource.AddFile( "materials/fwkzt/sprite_bloodspray5.vmt" )
+    resource.AddFile( "materials/fwkzt/sprite_bloodspray6.vmt" )
+    resource.AddFile( "materials/fwkzt/sprite_bloodspray7.vmt" )
+    resource.AddFile( "materials/fwkzt/sprite_bloodspray8.vmt" )
 end
 
-local function NoTTTGib( Ply )
-
-	local Head = Ply:LookupBone('valvebiped.bip01_head1')
-	if !Head then return end
-	local Pos = Ply:GetBonePosition( Head )
-
-	Ply:ManipulateBoneScale(Head, vector_origin)
-	
-	local ED = EffectData()
-		ED:SetEntity( Ply )
-		ED:SetNormal( -Ply:GetForward() )
-		ED:SetScale( Ply:EntIndex() )
-		ED:SetOrigin( Pos )
-	util.Effect( 'headshot', ED )
-	
+local function gibPlayerHead( Ply, Normal )
+    local Head = Ply:LookupBone( "valvebiped.bip01_head1" )
+    if not Head then return end
+    local Pos = Ply:GetBonePosition( Head )
+    local RagHead = Ply.server_ragdoll:LookupBone( "valvebiped.bip01_head1" )
+    if not RagHead then return end
+    Ply.server_ragdoll:ManipulateBoneScale( RagHead, Vector( 0, 0, 0 ) )
+    local ED = EffectData()
+    ED:SetEntity( Ply )
+    ED:SetNormal( Normal )
+    ED:SetScale( Ply.server_ragdoll:EntIndex() )
+    ED:SetOrigin( Pos )
+    util.Effect( "headshot", ED )
 end
 
-local function gibPlayerHead( Ply, Normal )
+-- Player Headshots
+local function PlayerDeath( Ply, _, Attacker )
+    if GAMEMODE_NAME ~= "terrortown" then return end
+    if not IsValid( Ply.server_ragdoll ) then return end
+    if not Ply.was_headshot then return end
+    if not IsValid( Attacker ) or not Attacker:IsPlayer() then return end
+    if not IsValid( Attacker:GetActiveWeapon() ) then return end
+    if Ply.IsGhost and Ply:IsGhost() then return end
 
-	local Head = Ply:LookupBone('valvebiped.bip01_head1')
-	if !Head then return end
-	local Pos = Ply:GetBonePosition( Head )
+    if Ply.OwnedBlackMarketItems and Ply.OwnedBlackMarketItems[CAT_EQUIPMENT] then

```

Comment:
shouldnt this be `Ply.OwnedBlackMarketItems[CAT_EQUIPMENT] == "fiber_helmet"`

Reply:
Good catch